src/shaders/MaterialX/vector4.h

```diff
@@ -338,7 +369,8 @@ vector4 min(vector4 in, float amount){
     return out;
 }
 
-vector4 fmod(vector4 in, vector4 amount){
+vector4 fmod(vector4 in, vector4 amount)
+{
     vector4 out = in;
```

Comment:
I see this idiom all over the place.

    T out = in;
    T out.x = ... // reassign
    T out.y = ... // reassign
    ...
    return out;

Why bother with the out=in if you're just going to write over it?

And for that matter, you don't need out at all, you can use constructors:

    return vector4 (fmod (in.x, amount.x),
                    fmod (in.y, amount.y),
                    fmod (in.z, amount.z),
                    fmod (in.w, amount.w),
);

I realize that what I'm suggesting is a pervasive and intrusive change. I am not asking that you do this as part of this checkin. Just something to think about for later.